api/hasura/auth.ts

```diff
@@ -39,6 +39,7 @@ export default async function handler(req: VercelRequest, res: VercelResponse) {
     res.status(200).json({
       'X-Hasura-User-Id': tokenRow.tokenable_id.toString(),
       'X-Hasura-Role': profile.admin_view ? 'superadmin' : 'user',
+      'X-Hasura-Address': profile.address,
```

Comment:
Yea this is good, we can use it like that. It can work for admin-secret too, just have to specify `X-Hasura-User-Id` in addition to the admin secret and it will allow the admin to "act like" a given user.